integration-tests/common.go

```diff
@@ -2,74 +2,212 @@ package integration_tests
 
 import (
 	"context"
-	"math/big"
+	"fmt"
 	"net/url"
 	"time"
 
-	"github.com/ethereum/go-ethereum/accounts/abi/bind"
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/go-resty/resty/v2"
+	. "github.com/onsi/gomega"
 	"github.com/rs/zerolog/log"
-	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
-	"github.com/smartcontractkit/chainlink-testing-framework/config"
-	"github.com/smartcontractkit/helmenv/environment"
+	uuid "github.com/satori/go.uuid"
+	"github.com/smartcontractkit/chainlink-env/environment"
+	"github.com/smartcontractkit/chainlink-env/pkg/helm/chainlink"
+	"github.com/smartcontractkit/chainlink-env/pkg/helm/mockserver"
+	mockservercfg "github.com/smartcontractkit/chainlink-env/pkg/helm/mockserver-cfg"
+	ops "github.com/smartcontractkit/chainlink-starknet/relayer/ops"
+	blockchain "github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	ctfClient "github.com/smartcontractkit/chainlink-testing-framework/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
 )
 
-// StarkNetNetworkConfig StarkNet network config
-type StarkNetNetworkConfig struct {
-	ContractsDeployed         bool          `mapstructure:"contracts_deployed" yaml:"contracts_deployed"`
-	L1BridgeAddr              string        `mapstructure:"l1_bridge_addr" yaml:"l1_bridge_addr"`
-	Name                      string        `mapstructure:"name" yaml:"name"`
-	ChainID                   int64         `mapstructure:"chain_id" yaml:"chain_id"`
-	URL                       string        `mapstructure:"url" yaml:"url"`
-	URLs                      []string      `mapstructure:"urls" yaml:"urls"`
-	Type                      string        `mapstructure:"type" yaml:"type"`
-	PrivateKeys               []string      `mapstructure:"private_keys" yaml:"private_keys"`
-	ChainlinkTransactionLimit uint64        `mapstructure:"chainlink_transaction_limit" yaml:"chainlink_transaction_limit"`
-	Timeout                   time.Duration `mapstructure:"transaction_timeout" yaml:"transaction_timeout"`
-	MinimumConfirmations      int           `mapstructure:"minimum_confirmations" yaml:"minimum_confirmations"`
-	BlockGasLimit             uint64        `mapstructure:"block_gas_limit" yaml:"block_gas_limit"`
+const (
+	gethClient      = "Hardhat"
+	devnetClient    = "starknet-dev"
+	chainLinkClient = "chainlink"
+	ChainName       = "starknet"
+	ChainId         = "devnet"
+)
+
+var (
+	err             error
+	Env             *environment.Environment
+	sc              *StarkNetClient
+	chainlinkNodes  []*client.Chainlink
+	mockServer      *ctfClient.MockserverClient
+	nodeKeys        []NodeKeysBundle
+	starknetNetwork *blockchain.EVMNetwork
+	bTypeAttr       *client.BridgeTypeAttributes
+)
+
+type StarkNetClient struct {
```

Comment:
I don't really understand the scope of this "client":

- It interacts with the CL node
- It interacts with the StarkNet node
- It also expects/asserts so acts as a test

We should have a `ChainlinkClient` a `StarkNetDevnetClient` and a test, instead of combining all of that functionality into one god object.

Reply:
God object decoupled